core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/model/VirtualDocument.scala

```diff
@@ -138,4 +138,14 @@ abstract class VirtualDocument[T] extends ReadonlyVirtualDocument[T] {
 
   def getTotalFileSize: Long =
     throw new NotImplementedError("getTotalFileSize method is not implemented")
+
+  /**
+    * The following two methods are used to directly access files in the underlying storage system.
+    * For example, in Iceberg, it provides direct access to Parquet files.
+    */
+  def getUnderlyingFileCount: Int = 0
```

Comment:
The below two functions are not generalized enough, so it shouldn't be added to the VirtualDocument.

You can implement asInputStream for the IcebergDocument. And in the implementation, you create the Zip stream. I think you don't need to differentiate the case of one parquet or multiple parquet. It is fine to use Zip to handle all these cases.

Once you change this part, you also need to update the ResultExportService